SFML/SFML
# [Feature] Adding constexpr functionality to the library

* [X] Has this change been discussed on [the forum](https://en.sfml-dev.org/forums/index.php#c3) or in an issue before? Discussed in Discord.

----

## Description

Added constexpr support for a few classes (more will be added later).

It's important to note that constexpr functionality cannot be buil separately and then linked with the rest of the library. This is because the compiler must be able to see the definition of constexpr functionality in order to perform compile-time computations.

For that reason, definitions of constexpr functionality must be done inside the header file (just like templates), so many definitions that used to be in the corresponding `.cpp` file are now inside a `.inl` file (to be consistent with the rest of the library).

Another discussion is about turning some variables that are currently const into constexpr. This should be easy in general, but static class variables which type is the same as the class they're in (e.g.: `sf::Color::Blue`, `sf::Color::Red`, `sf::Time::Zero`, ...) cannot be made constexpr (at least I'm not aware of how to do it).

For the moment I'm leaving all the variable definitions inside the `.cpp` file.

## Tasks


## How to test this PR?

This is the template example, but using a constexpr `sf::Vector2u` to intialize `sf::VideoMode` instead of using the values directly. The constexpr functionality should work out of the box, as long as the initialization is performed at compile time.

```cpp
#include <SFML/Graphics.hpp>
#include <SFML/System.hpp>

int main()
{
    // Just showing sf::Vector2, but should work for every constexpr functionality
    constexpr sf::Vector2u windowSize(1280, 720);
    sf::RenderWindow window(sf::VideoMode(windowSize.x, windowSize.y), "Minimal, complete and verifiable example");
    window.setFramerateLimit(60);

    while (window.isOpen())
    {
        sf::Event event;
        while (window.pollEvent(event))
        {
            if (event.type == sf::Event::Closed)
                window.close();
        }

        window.clear();
        window.display();
    }
}
```
